### PR TITLE
Update with new type specs

### PR DESCRIPTION
**Pull request by philschanely**
- [x] Work in updated type specs

### Testing

Open the app and browse around to see elements are still as they were before.

### CODE REVIEW

File: app/views/sage/examples/objects/nav/_preview.html.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
 <div class="sage-nav">
   <a hre="#" class="sage-nav__link">Nav Link</a>
```
Comment (teenwolfblitzer) [inline app/views/sage/examples/objects/nav/_preview.html.erb]: looks like the `href` is missing the `f` here and below.

File: app/assets/stylesheets/sage/system/tokens/_color_palette.scss [hunk 2]
```diff
@@ -60,4 +60,11 @@ $sage-colors: (
     400: #304050,
     500: #263240,
   ),
+  white: (
+    100: #fff,
```
Comment (teenwolfblitzer) [inline app/assets/stylesheets/sage/system/tokens/_color_palette.scss]: 🏳️

File: app/assets/stylesheets/sage/system/patterns/elements/_switch.scss [hunk 3]
```diff
@@ -6,116 +6,106 @@
...
-  appearance: none;
-  color: $sage-switch-color-default;
-  background: $sage-switch-color-default;
-  border-radius: $sage-switch-border-radius;
-  outline: none;
-  transition: background 0.3s ease-out;
-  cursor: pointer;
-
-  + .sage-switch__label {
+  input[type="checkbox"] {
```
Comment (teenwolfblitzer) [inline app/assets/stylesheets/sage/system/patterns/elements/_switch.scss]: Looks like this change reverted a recent update, using the class over the element selector (screen below of unstyled radio button)

switch-radio

File: app/assets/stylesheets/sage/system/patterns/elements/_switch.scss [hunk 4]
```diff
@@ -6,116 +6,106 @@
...
+      background: sage-color(white);
+      border-radius: sage-border(radius-round);
+      box-shadow: 0 1px 2px rgba(0, 0, 0, 0.18), 0 1px 3px rgba(0, 0, 0, 0.1);
+      transition: transform 0.2s ease-in-out;
     }
-  }
 
-  &:invalid {
-    background-color: sage-color(red);
+    // checked
```
Comment (teenwolfblitzer) [inline app/assets/stylesheets/sage/system/patterns/elements/_switch.scss:77]: Invalid/error state missing here

File: app/assets/stylesheets/sage/system/patterns/elements/_form_input.scss [hunk 5]
```diff
@@ -15,22 +15,22 @@
...
+  transform: translateY(calc(#{$sage-body-font-size} * -1));
   left: $sage-inputfield-padding;
   top: 50%;
   color: inherit;
-  font-size: $sage-inputfield-label-font-size; // 11px
-  font-weight: $sage-inputfield-label-font-weight;
   pointer-events: none;
   opacity: 0;
+
+  @extend %t-sage-body-xsmall-semi;
```
Comment (teenwolfblitzer) [inline app/assets/stylesheets/sage/system/patterns/elements/_form_input.scss:25]: 🚀

**Review by teenwolfblitzer — CHANGES_REQUESTED** [review]
One issue to look at before 👍, seems a couple of the recent switch changes didn't get pulled through completely.

File: app/assets/stylesheets/sage/system/patterns/elements/_switch.scss (hunk 3 above)
Comment (philschanely) [reply app/assets/stylesheets/sage/system/patterns/elements/_switch.scss]: This must have been a merge mishap; I'd definitely have preferred the BEM name here...

**Review by teenwolfblitzer — APPROVED** [review]
👍 LGTM, nicely done!

**Pull request merged**